docs/src/main/asciidoc/mongo-guide.adoc

```diff
@@ -0,0 +1,599 @@
+
+If you want to use Docker to run a Mongo database, you can use the following command to launch one:
+[source,shell]
+----
+docker run -ti --rm -p 27017:27017 mongo:4.0
+----
+
+== Creating a frontend
+
+Now let's add a simple web page to interact with our `FruitResource`.
+Quarkus automatically serves static resources located under the `META-INF/resources` directory.
+In the `src/main/resources/META-INF/resources` directory, add a `fruits.html` file with the content from this {quickstarts-blob-url}/rest-json/src/main/resources/META-INF/resources/fruits.html[fruits.html] file in it.
+
+You can now interact with your REST service:
+
+ * start Quarkus with `./mvnw compile quarkus:dev`
+ * open a browser to `http://localhost:8080/fruits.html`
+ * add new fruits to the list via the form
+
+== Reactive Mongo Client
+A reactive Mongo Client is included in Quarkus.
+Using it is as easy as using the classic Mongo Client. You can rewrite the previous example to use it like the following.
+
+[source,java]
+----
+package org.acme.rest.json;
+
+import io.quarkus.mongo.ReactiveMongoClient;
+import io.quarkus.mongo.ReactiveMongoCollection;
+import org.bson.Document;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+@ApplicationScoped
+public class ReactiveFruitService {
+
+    @Inject
+    ReactiveMongoClient mongoClient;
+
+    public CompletionStage<List<Fruit>> list(){
+        return getCollection().find().map(doc -> {
+            Fruit fruit = new Fruit();
+            fruit.setName(doc.getString("name"));
+            fruit.setDescription(doc.getString("description"));
+            return fruit;
+        }).toList().run();
+    }
+
+    public CompletionStage<Void> add(Fruit fruit){
+        Document document = new Document()
+                .append("name", fruit.getName())
+                .append("description", fruit.getDescription());
+        return getCollection().insertOne(document);
+    }
+
+    private ReactiveMongoCollection<Document> getCollection(){
+        return mongoClient.getDatabase("fruit").getCollection("fruit");
+    }
+}
+----
+
+
+[source,java]
+----
+package org.acme.rest.json;
+
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+@Path("/reactive_fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class ReactiveFruitResource {
+
+    @Inject ReactiveFruitService fruitService;
+
+
+    @GET
+    public CompletionStage<List<Fruit>> list() {
+        return fruitService.list();
+    }
+
+    @POST
+    public CompletionStage<List<Fruit>>  add(Fruit fruit) {
+        fruitService.add(fruit);
+        return list();
+    }
+}
+----
+
+== Simplifying Mongo Client usage
```

Comment:
Same comment about MongoDB but I suppose it depends on how you want to name the extension in the end.

Reply:
Here as it's not followed with database, it should be MongoDB.